common/web/types/src/util/util.ts

```diff
@@ -15,3 +15,13 @@ export function boxXmlArray(o: any, x: string): void {
     }
   }
 }
+
+const MATCH_HEX_ESCAPE = /\\u{([0-9a-fA-F]{1,5})}/g;
```

Comment:
```suggestion
const MATCH_HEX_ESCAPE = /\\u{([0-9a-fA-F]{1,6})}/g;
```

`\u{10FFFF}` has 6 digits.

Query: `\u{a}` doesn't seem like it should be acceptable? Do we want to allow control characters (below 0x20) in these strings? This perhaps should be handled outside this function though.